server/catgenome/src/main/java/com/epam/catgenome/manager/reference/MotifSearchManager.java

```diff
@@ -274,12 +297,16 @@ private Chromosome getNextChromosome(final List<Chromosome> chromosomes, final C
         return null;
     }
 
-    private Chromosome loadChrById(final Long referenceId, final Long chromosomeId) {
+    private Chromosome fetchChromosomeById(final Reference reference, final Long chromosomeId) {
         if (chromosomeId == null) {
-            return getFirstChromosomeFromGenome(referenceId);
-        } else {
-            return referenceGenomeManager.loadChromosome(chromosomeId);
+            return getFirstChromosomeFromGenome(reference.getId());
```

Comment:
do we need to load first chr if we already have all in reference object? can we just get first one?